### PR TITLE
PSBT: add no_publish flag for safe batch channel opens v2

### PR DESCRIPTION
**Pull request by cfromknecht**
To make it safer to open multiple channels in one batch, we add the `--no_publish` flag to `lncli openchannel` (called `no_publish` in the `PsbtShim` in RPC).

When opening multiple channels in the same transaction, the `--no_publish` flag should be set for every channel but the last one. That way only after the last channel is negotiated successfully the batch transaction is published to the network. If that isn't done, it could lead to loss of funds if one of the peers involved times out and no commitment transactions were signed for that channel.

Shepherding #4417 since @guggero is offline for now.

### CODE REVIEW

File: cmd/lncli/cmd_open_channel.go [hunk 1]
```diff
@@ -163,6 +163,15 @@ var openChannelCommand = cli.Command{
 				"as a base and add the new channel output to " +
 				"it instead of creating a new, empty one.",
 		},
+		cli.BoolFlag{
+			Name: "no_publish",
+			Usage: "when using the interactive PSBT mode to open " +
+				"multiple channels in a batch, this flag " +
+				"instructs lnd to not publish the full batch " +
+				"transaction just yet. for safety reasons " +
```
Comment (Roasbeef) [inline cmd/lncli/cmd_open_channel.go:171]: Style nit: usage contains a period, but what follows isn't capitalized.

File: docs/psbt.md [hunk 2]
```diff
@@ -231,3 +231,62 @@ Base64 encoded PSBT: cHNidP8BAH0CAAAAAbxLLf9+AYfqfF69QAQuETnL6cas7GDiWBZF+3xxc/Y
...
+
+Combined command to get batch PSBT:
+```bash
+bitcoin-cli walletcreatefundedpsbt [] '[{"tb1q53626fcwwtcdc942zaf4laqnr3vg5gv4g0hakd2h7fw2pmz6428sk3ezcx":0.01000000},{"tb1qywvazres587w9wyy8uw03q8j9ek6gc9crwx4jvhqcmew4xzsvqcq3jjdja":0.01000000}]'
+```
+
+### Safety warning about batch transactions
+
+As mentioned before, the PSBT channel funding flow works by pausing the funding
+negotiation with the remote peer directly after the multisig key have been
```
Comment (yyforyongyu) [inline docs/psbt.md:264]: typo
```suggestion
negotiation with the remote peer directly after the multisig key has been
```
or
```suggestion
negotiation with the remote peer directly after the multisig keys have been
```

**Comment by yaslama** [comment]
@cfromknecht  There is a use case for the --no_publish flag even outside the context of PSBT: when using "zeroconf" channels (#4424).
Alice wants to pay to Carole, but Carole has no channel with sufficient capacity. So she sends an invoice to Alice with routing hints with a dummy channel between Bob and Carole. When Bob receives the HTLCs of the payment from Alice, he creates a zeroconf channel and forward the HTLCs to Carole. The problem is that there are cases when only part of the HTLCs reaches Bob, so Bob wants to "really" open the channel only when all the amount was forwarded to Carole. On way to do that is to open the channel with a "--no_publish" option and publish it only after all the HTLCs are forwarded to Carole.
So what do you think about extending the usage --no-publish for all channel creation.

File: docs/psbt.md (hunk 2 above)
Comment (cfromknecht) [reply docs/psbt.md]: fixed!

**Comment by Roasbeef** [comment]
> So what do you think about extending the usage --no-publish for all channel creation.

Seems useful, but out of the scope of this PR, as we just want to modify the existing behavior.

**Comment by cfromknecht** [comment]
cc @joostjager PTAL

**Pull request merged**